locations/spiders/essentiel_antwerp.py

```diff
@@ -1,30 +1,48 @@
-import re
+from typing import Any
 
-import scrapy
+from scrapy.http import Response
+from scrapy.linkextractors import LinkExtractor
+from scrapy.spiders import CrawlSpider, Rule
 
+from locations.google_url import extract_google_position
+from locations.hours import DAYS_FULL, OpeningHours
 from locations.items import Feature
+from locations.pipelines.address_clean_up import merge_address_lines
 
 
-class EssentielAntwerpSpider(scrapy.Spider):
+class EssentielAntwerpSpider(CrawlSpider):
     name = "essentiel_antwerp"
     item_attributes = {"brand": "Essentiel Antwerp", "brand_wikidata": "Q117456339"}
-    start_urls = ["https://www.essentiel-antwerp.com/be_fr/storelocator"]
+    start_urls = ["https://www.essentiel-antwerp.com/be_en/ourstores"]
 
-    def parse(self, response):
-        data = response.text
-        stores = json.loads(re.search(r"stores:(\[.*\])", data).group(1))
-        for store in stores:
-            item = Feature()
-            item["ref"] = store["storelocator_id"]
-            item["phone"] = store["phone"]
-            item["country"] = store["country"]
-            item["state"] = store["state"]
-            item["postcode"] = store["zipcode"]
-            item["lat"] = float(store["latitude"])
-            item["lon"] = float(store["longtitude"])
-            item["street_address"] = store["address"]
-            item["city"] = store["city"]
-            item["name"] = store["name"]
-            item["website"] = f'https://www.essentiel-antwerp.com/eu/{store["rewrite_request_path"]}/'
-            yield item
+    rules = [
+        Rule(
+            LinkExtractor(
+                restrict_xpaths='//*[@class="sps-NpIex BuilderAccordion_accordion_item__pPRwV"]//*[@class="sps-C92Xj"]//a'
+            ),
+            callback="parse",
+        )
+    ]
+
+    def parse(self, response: Response, **kwargs: Any) -> Any:
+        item = Feature()
+        item["street_address"] = response.xpath("//p/text()").get()
+        item["addr_full"] = merge_address_lines(
+            [item["street_address"], response.xpath("//p[2]/text()").get(), response.xpath("//p[3]/text()").get()]
+        )
+        item["website"] = item["ref"] = response.url
+        item["phone"] = response.xpath('//*[contains(@href,"tel:")]//text()').get()
+        extract_google_position(item, response)
+        item["opening_hours"] = OpeningHours()
+        for time_string in response.xpath('(//*[@class="ImageBlock_paragraph__43zjD"])[3]//p'):
+            time = time_string.xpath(".//text()").get()
+            if time in ["Closed", None]:
+                continue
+            if "&" in time:
+                for open_close_time in time.split("&"):
+                    open_time, close_time = open_close_time.split("-")
+            else:
+                open_time, close_time = time.split("—")
+            for day in DAYS_FULL:
+                item["opening_hours"].add_range(day=day, open_time=open_time.strip(), close_time=close_time.strip())
```

Comment:
This needs fixing or removing.

Reply:
@Cj-Malone  I'll try to fix bug if it's simpler else I'll remove the opening_hours as per your suggestion.

Reply:
@mohit-sharma2194 in this case given the small number of POIs, don't waste time trying to fix. Just delete thanks.